src/LibraryViewExtension/LibraryViewCustomization.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Dynamo.Wpf.Interfaces;
+
+namespace Dynamo.LibraryUI
+{
+    class LibraryViewCustomization : ILibraryViewCustomization
+    {
+        public const string DefaultSectionName = "default";
+        private LayoutSpecification root = new LayoutSpecification();
+
+        private void RaiseSepcificationUpdated()
+        {
+            var handler = SpecificationUpdated;
+            if (handler != null) handler(this, EventArgs.Empty);
+        }
+
+        /// <summary>
+        /// Gets a copy of the current library layout specification
+        /// </summary>
+        public LayoutSpecification Specification
+        {
+            get
+            {
+                return root.Clone();
```

Comment:
Why is a copy returned? Usually for property get methods, I have never seen cases where a copy is returned. If you really want to return a copy, it is better to add a specific method for that.